File: BE/models/Location.js
/**
 * Model đại diện cho địa điểm đi chơi
 */
export class Location {
  constructor(data = {}) {
    this.id = data.id || null;
    this.name = data.name || '';
    this.address = data.address || '';
    this.description = data.description || '';
    this.locationTypeId = data.locationTypeId || null;
    this.imageUrl = data.imageUrl || '';
    this.rating = data.rating || 0;
    this.phone = data.phone || null;
    this.website = data.website || null;
  }
}

File: BE/src/main/java/com/food/app/model/Location.java
package com.food.app.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Model đại diện cho địa điểm đi chơi
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Location {
    private Long id;
    private String name;
    private String address;
    private String description;
    private Long locationTypeId; // Tham chiếu đến LocationType
    private String imageUrl; // URL ảnh địa điểm
    private Double rating; // Đánh giá trung bình (0-10)
    private String phone;
    private String website;
}


